src/riscv/lib/src/jit/builder/instr_map.rs

```diff
@@ -0,0 +1,144 @@
+use std::fmt::Debug;
+
+use crate::jit::builder::instruction::LoweredInstruction;
+use crate::jit::builder::sequence::sequence_analysis::InstrInfo;
+use crate::machine_state::memory::Address;
+
+/// A holder of the Addresses of the instructions in a sequence.
+#[derive(Debug, Clone)]
+pub(super) struct AddrMap {
+    addresses: Vec<Address>,
+}
+
+impl AddrMap {
+    /// Create a new, empty AddrMap.
+    pub(super) fn new() -> Self {
+        Self {
+            addresses: Vec::new(),
+        }
+    }
+
+    /// Insert a new address into the map.
+    pub(super) fn insert(&mut self, address: Address) {
+        self.addresses.push(address);
+    }
```

Comment:
The `InstrMapBuilder` should be in charge of creating this. The type should not allow mutation after creation. So I think these methods should not exist.